packages/starlight/style/props.css

```diff
@@ -87,8 +87,8 @@
 		'Segoe UI Symbol', 'Noto Color Emoji';
 	--sl-font-system-mono: ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, 'Liberation Mono',
 		'Courier New', monospace;
-	--__sl-font: var(--sl-font, ''), var(--sl-font-system);
-	--__sl-font-mono: var(--sl-font-mono, ''), var(--sl-font-system-mono);
+	--__sl-font: var(--sl-font-system);
+	--__sl-font-mono: var(--sl-font-system-mono);
```

Comment:
This change would break the intended behaviour here: that a user can set `--sl-font` and/or `--sl-font-mono` to apply a custom font, while still preserving the fallback system font stack Starlight defines.

In the existing code, this worked (on most devices 😅), because when those two variables weren’t set, you’d end up with font stacks something like this (if I expand the variables):
```css
--__sl-font: '', ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, 'Helvetica Neue', Arial, 'Noto Sans', sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
--__sl-font-mono: '', ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, 'Liberation Mono', 'Courier New', monospace;
```

The `''` value is ignored usually, then as soon as `--sl-font` and `--sl-font-mono` are set — let’s say to imaginary `Astro Sans` and `Astro Mono` fonts — this switches to:
```css
--__sl-font: 'Astro Sans', ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, 'Helvetica Neue', Arial, 'Noto Sans', sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
--__sl-font-mono: 'Astro Mono', ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, 'Liberation Mono', 'Courier New', monospace;
```

I was thinking how to keep the current behaviour while avoiding the risk of hitting an unintended font, which is apparently possible with the `''` value in your case, and I think we can do this instead:

```suggestion
	--__sl-font: var(--sl-font, var(--sl-font-system)), var(--sl-font-system);
	--__sl-font-mono: var(--sl-font-mono, var(--sl-font-system-mono)), var(--sl-font-system-mono);
```

This might look a bit odd, but in the fallback case, where no user-defined fonts are used, we basically double up the system font stack. The second value is effectively pointless until a user sets a custom font, and then the second value makes sure the system fallbacks are still in place.

Reply:
@delucis thanks for taking the time to look at this PR, It's definitely and odd behavior because it works with chrome and edge without an issue, someone else had the same issue here : [https://github.com/withastro/astro.build/issues/943](Weird font rendering on Firefox ) 

at first i didn't understand the use of `--sl-font` because on my branch that variable didn't exist and that's why i ignored it, but now with your explanation i understand why, it's user specific font preference, so we need to keep it as the first option and fall back on the `--sl-font-system` (same with mono)

i put in place your suggested fix in a new commit